### PR TITLE
1280: Simplified planning form. Added default value

### PR DESCRIPTION
**Pull request by tuj**
#### Link to ticket

https://leantime.itkdev.dk/#/tickets/showTicket/1280

#### Description

Simplified planning form. Added default value.

#### Checklist

- [x] My code is covered by test cases.
- [x] My code passes our test (all our tests).
- [x] My code passes our static analysis suite.
- [x] My code passes our continuous integration process.

### CODE REVIEW

File: assets/controllers/planning-scroll_controller.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        const firstColumn = this.columnTargets.find(
+            (el) => el.dataset.index == 1,
+        );
+
+        if (!scrollToColumn || !firstColumn) {
+            return;
+        }
+
+        const { x } = scrollToColumn.getBoundingClientRect();
+        const { x: firstColumnX } = firstColumn.getBoundingClientRect();
```
Comment (rimi-itk) [inline assets/controllers/planning-scroll_controller.js:28]: It took me a while to actually understand what's going on here.

```suggestion
        const x = scrollToColumn.getBoundingClientRect().x; 
        const firstColumnX = firstColumn.getBoundingClientRect().x;
```

is much easier to understand.

File: src/Service/LeantimeApiService.php [hunk 2]
```diff
@@ -378,6 +377,7 @@ public function getPlanningDataWeeks(): PlanningData
         $weeks = $planning->weeks;
 
         $currentYear = (int) (new \DateTime())->format('Y');
+        $currentWeek = (int) (new \DateTime())->format('W');
 
         for ($weekNumber = 1; $weekNumber <= 52; ++$weekNumber) {
```
Comment (rimi-itk) [inline src/Service/LeantimeApiService.php:382]: A year can actually have 53 weeks (cf. https://en.wikipedia.org/wiki/ISO_week_date#Weeks_per_year), and the year 2026 (in the near future is one of them).

**Review by rimi-itk — APPROVED** [review]
Approved with a couple of comments.

File: assets/controllers/planning-scroll_controller.js (hunk 1 above)
Comment (yepzdk) [reply assets/controllers/planning-scroll_controller.js]: Changed it. Thanks for the suggestion.

File: src/Service/LeantimeApiService.php (hunk 2 above)
Comment (yepzdk) [reply src/Service/LeantimeApiService.php:384]: I can't take credit for this. So, I have pasted your suggestion into a comment.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**